### ISSUE TITLE
Motion Tracking Issue (always returns to 0,0 and not last position)

### ISSUE THREAD

**Issue by SiliconAvatar** [EliasKotlyar/Xiaomi-Dafang-Hacks#802]
I have an issue with my WyseCam Pan camera running Dafang-Hacks, where when motion tracking is enabled, and the "Motion Timeout" is set to a positive number (IE 15) the camera always returns to the bottom left position (0,0) instead of the last known position as it should.

The Motion Timeout setting is described as "Restore camera position after x seconds, -1 to deactivate" which I take to mean that it will track motion, then after there is no more motion, it will return to it's original coordinates. It does not currently do that.

I think I've narrowed down the issue a bit.

There seems to be at least 3 separate issues in play here as far as I can tell, all seem to be in detectionTracking.sh.

First: The script detectionTracking.sh is meant to store the camera's original position to /tmp/cameraposition (line 16) however it never creates this file if it does not exist. I've tried this multiple times while SSH'ed into it watching that folder, and the file "cameraposition" never gets created. If I run "touch cameraposition" inside the /tmp/ folder, the script then starts storing the coordinates in that file as it should. There the logic to create the file if it does not already exist should be revised.  

Second: The script seems to store the camera's position every time there is no motion detected (Line 103). This is a problem. For example, if I have the motion timeout set to 15 seconds, the intention would be for the camera to sit at the "home" position, wait for motion, then track said motion. After no more motion after 15 seconds, it returns to 'home'. The problem is that every time the scripts stops seeing motion, it over-writes the 'home' position with where it was last at when the motion stopped. This will lead to the home position gradually walking away from where it was originally set to. I would think that this could be fixed by only writing the original position to the file at the start of a motion event, then not overwriting it until the entire motion event is over (taking the timeout into consideration) and already returned to base. Alternatively, the home position could be set once, by the user, then never overwritten.

Third: The script gets the camera's position by calling "motor -d s" (Line 104). However if the camera is currently traveling when this is called, it will return 0 for both the X and Y axis. There needs to be a sleep in here or something so that this is only called once the camera is stationary. 

I'm not enough of a programmer to know how to fix these, but I think my observations were correct. If I got something wrong, please let me know.

This is also likely a duplicate of Issue#567 and Issue#652, so it seems like a problem at least a few users are seeing.

**Comment by jmtatsch** [comment]
Your observations make sense. A pull request to fix these would be very welcome.

**Comment by beckynet** [comment]
I've also trouble with Motion Tracking.
When motion is detected to the right of the picture the Camera turns to the left.
The vertical movement also works upside down, when motion is captured in the bottom of the picture the sensor goes up.

**Comment by nerdosity** [comment]
After seeing my camera lookin to the wall for days and days, I fixed it by editing the detectionTracking.sh.
After a calibration, then making my camera "looks good", then creating the cameraposition file, I get the "right" position to look at.
I just inserted it into the detectionTracking.sh script and everything is working fine now.
Why this is not implemented in the UI? I think it's advisable to let users point the camera in the "right" direction, not giving too much head to automatic motion follow.
